OrbitLinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -2,6 +2,40 @@
 
 namespace LinuxTracing {
 
+void UprobesFunctionCallManager::ProcessUprobes(pid_t tid,
+                                                uint64_t function_address,
+                                                uint64_t begin_timestamp) {
+  auto& tid_timer_stack = tid_timer_stacks_[tid];
+  tid_timer_stack.emplace(function_address, begin_timestamp);
+}
+
+std::optional<FunctionCall> UprobesFunctionCallManager::ProcessUretprobes(
+    pid_t tid, uint64_t end_timestamp) {
+  if (tid_timer_stacks_.count(tid)) {
+    auto& tid_timer_stack = tid_timer_stacks_.at(tid);
+
+    // This is because we erase the stack for this thread as soon as it becomes
+    // empty.
+    assert(!tid_timer_stack.empty());
+
+    if (!tid_timer_stack.empty()) {
```

Comment:
If you keep the "if" despite the "assert", also early out here:

if (tid_timer_stack.empty()) {
  return std::optional<FunctionCall>{};
}

// Now comes the real body of the function.